### ISSUE TITLE
Allow 'In Stock' to override 'To Order' checkbox

### ISSUE THREAD

**Issue by lin-d-hop** [openfoodfoundation/openfoodnetwork#1228]
When a product is checked 'To Order' the 'In Stock' quantity is set to infinity to allow the enterprise to supply this product 'To Order'.

This functionality works fine but causes some user confusion when the 'To order' box is checked, then the user inputs an 'In Stock' quantity. The 'To Order' checkbox overrides the 'In stock' quantity and the value remains at Infinity.

A simple UX improvement would be set the 'To Order' checkbox to unchecked when the user inputs a value into the 'In Stock' field.

Note - Only 1.1 is a change to current functionality.

1. As an enterprise user I input a value into the 'In Stock' field:
 1.1 _If the 'To Order' box is checked -> the box is set to unchecked._
 1.2 If the 'To Order' box is unchecked -> no change
2. As an enterprise user I check the 'To Order' checkbox:
 2.1 If the 'In Stock' field contains a number -> the field is set to Infinity
 2.2 If the 'In Stock' field is set to Infinity -> no change

![instocktoorder](https://cloud.githubusercontent.com/assets/8669848/20571991/86034b18-b1a1-11e6-9daa-e7d62ee43fa8.png)

**Comment by Matt-Yorkley** [comment]
I can implement something for this in the next couple of days. I think a bit of javascipt that updates the form as the user is interacting with it would be good, and quite easy to do. I jotted down some notes for it. 

Maybe as a separate issue, but relating to this one: it might be good in a few places where the UI is a bit vague to implement some sort of tooltip functionality for some of the more confusing parts of the backend. Maybe a question mark with a rollover that gives a short explanation.

**Comment by lin-d-hop** [comment]
@Matt-Yorkley UX is being reviewed. This issue is small but many more to come. Let's keep issues to issues and not blow them up to a whole vague UX scope eh?

**Comment by Matt-Yorkley** [comment]
Fair point, I was just trying to be thorough as the issue was largely about UX and lack of clarity. I've added a "what's this?" help tooltip as well as the UX improvements to the form. Pull request submitted and ready for testing.

**Comment by lin-d-hop** [comment]
@Matt-Yorkley Could you explore adding some specs to capture this behaviour? This seems like a change that should have test coverage.

**Comment by lin-d-hop** [comment]
@Matt-Yorkley Ignore above, was on the wrong issue (on my phone on a train in Germany = hard to navigate). Test coverage less important here me thinks.

**Comment by lin-d-hop** [comment]
@Matt-Yorkley Failing tests...

**Comment by Matt-Yorkley** [comment]
@lin-d-hop ready for testing, in a new branch with uk prefix.

**Comment by OliverUK** [comment]
Tested now on UK staging @lin-d-hop @Matt-Yorkley 

Here is how it behaves in edit product:
- If the To Order box is ticked, the quantity field cannot be edited and is set to infinity.
- If the box isn't ticked the quantity can be edited.

In **create** product no such thing happens. In fact I've been recently noticing that if you tick "on demand" in the process of creating a product, it saves it as zero stock and has to be corrected in BEP. I'll raise a separate issue for this.

In BEP there is no change to how it behaves. I can enter a quantity and then also tick the "on demand" box and both will show at the same time, at least until the screen is refreshed. Similar behaviour in Inventory.

I also think that we should either use "on demand" or "to order" or even better just call it something like "unlimited" but certainly not a mix of different terms for the same thing. :)

**Comment by lin-d-hop** [comment]
Great, thanks @OliverUK 
So from this... 
a) Please create an issue for create product and add to UK Current milestone, dev ready pipeline.
b) Create an issue regarding 'on demand' / 'to order' and with same milestone/pipe and assign to yourself. This is a translation change and I'm looking forward to teaching you how to do this.
c) This fix itself looks good so once the above is done let's move to 'done' column (you have permission to do this now).

**Comment by sstead** [comment]
Looks good. The edit product interface is improved. New, BEP and Inventory unchanged.
